osquery/dispatcher/scheduler.h

```diff
@@ -43,6 +44,45 @@ class SchedulerRunner : public InternalRunnable {
   std::chrono::milliseconds getCurrentTimeDrift() const noexcept;
 
  private:
+   inline void calculateTimeDrift(std::chrono::time_point<std::chrono::steady_clock> start_time_point) {
+     auto loop_step_duration =
+         std::chrono::duration_cast<std::chrono::milliseconds>(
+             std::chrono::steady_clock::now() - start_time_point);
+     if (loop_step_duration + time_drift_ < interval_) {
+       pauseMilli(interval_ - loop_step_duration - time_drift_);
+       time_drift_ = std::chrono::milliseconds::zero();
+     } else {
+       time_drift_ += loop_step_duration - interval_;
+       if (time_drift_ > max_time_drift_) {
+         // giving up
+         time_drift_ = std::chrono::milliseconds::zero();
+       }
+     }
+   }
+
+   inline void maybeRunDecorators(size_t i) {
+     // Configuration decorators run on 60 second intervals only.
+     if ((i % 60) == 0) {
+       runDecorators(DECORATE_INTERVAL, i);
+     }
+   }
+
+   inline void maybeReloadSchedule(size_t i, uint64_t schedule_reload, bool schedule_reload_sql) {
+     if (schedule_reload > 0 && (i % schedule_reload) == 0) {
+       if (schedule_reload_sql) {
+         SQLiteDBManager::resetPrimary();
+       }
+       resetDatabase();
+     }
+   }
+
+   inline void maybeFlushLogs(size_t i) {
```

Comment:
Why do you need to `inline` all of this functions? Do we have a notable benefits to use inline here instead of common definitions in `.cpp`?

As far as C++ inline is a tricky mechanism to solve performance problems, this inlines looks like preliminary optimisation. I'd suggest to use common definitions in `*.cpp` file. And may be inline it later if we need it.

Reply:
I see your point, for sure. But without performance measurements `inline`s look a bit pointless.

Reply:
Without inlining, we'd need to take references to i and other parameters in order to avoid reallocating them on the stack.
Maybe the compiler is able to inline these helpers and maybe not, I don't really have the time to check.
Does it really matter?
We're 100% positive right now that we're not hurting the performance of the scheduler since the emitted code is exactly the same.

Reply:
Yes, it does matter. Because it is not desirable to the interface of module too verbose without a solid reason to. Please, in terms of this PR, please put implementation in `.cpp` file.